### ISSUE TITLE
Update a single row

### ISSUE THREAD

**Issue by benjeffery** [tskit-dev/tskit#1545]
### Discussed in https://github.com/tskit-dev/tskit/discussions/1543

<div type='discussions-op-text'>

<sup>Originally posted by **hyanwong** June 30, 2021</sup>
At the moment I think adding to the metadata arrays (e.g. on populations, individuals, or nodes) is quite intricate, and I have to look it up every time. I think it's something like this:

```
ts = msprime.sim_ancestry(10)
tables = ts.dump_tables()
pop_metadatas = [p.metadata for p in ts.populations()]  # Why is there no tables.population.get_metadatas() method?
pop_metadatas[0]['name'] += "(the only population)"
tables.populations.packset_metadata(
    [tables.populations.metadata_schema.validate_and_encode_row(r) for r in pop_metadatas])
new_ts = tables.tree_sequence()
print([p for p in new_ts.populations()])
```

(it looks me about 10 minutes to look up how to do that, and the information is scattered all over the docs - I don't think a beginner could do it TBH).

Should we have some convenience methods to help with this? I quite often want to edit metadata on just a single entry in a table: e.g. label a particular individual, flag up a single node, etc.

We should also give an example of doing this in the [metadata tutorial](https://tskit.dev/tutorials/metadata.html)</div>

**Comment by benjeffery** [comment]
I've added this to the C 1.0 milestone as we'll need something like: `tsk_update_char_ragged_array(tsk_size_t *offset_to_modify, char *array_to_modify, tsk_size_t *row_indexes, tsk_size_t *new_offset, char *new_array)`

**Comment by jeromekelleher** [comment]
I'm not sure this is something we want in the public API - wouldn't it be better to support updating a single row? If people are going to be updating multiple rows, then they should work column wise anyway.

**Comment by benjeffery** [comment]
Gah, of course, forgetting the C milestone is for the public API. Was thinking this would just be used by the CPython code.

**Comment by jeromekelleher** [comment]
Mind if we generalise this to "update single row" @benjeffery? No point in just focusing on the ragged columns here, we want an ``x_table_set_row`` as discussed [here](https://github.com/tskit-dev/tskit/discussions/1543#discussioncomment-942289)

**Comment by jeromekelleher** [comment]
Removing C API tag as it's done there.

**Comment by benjeffery** [comment]
The proposed syntax here is `tables.nodes[0] = tables.nodes[0].replace(flags=tskit.NODE_IS_SAMPLE)` where the RHS can be any object that has the neccesary attributes for the table on the LHS.

**Comment by jeromekelleher** [comment]
Can this be closed now?

**Comment by benjeffery** [comment]
Closed by #1600
